index.html

```diff
@@ -589,144 +457,36 @@ <h2>Attributes</h2>
-              the background.</p>
-            </li>
-            <ol>
-              <li>
-                <p>Send <var>data</var> in a datagram.</p>
-              </li>
-              <li>
-                <p>When an ack is received for the sent datagram,
-                <a>resolve</a> <var>p</var> with <code>true</code>.</p>
-              </li>
-              <li>
-                <p>When the datagram is detemined to be lost, <a>resolve</a>
-                <var>p</var> with <code>false</code>.</p>
-              </li>
-            </ol>
+            <!-- TODO: Figure out how to indicate to JS that a datagram has been acked  
+                       or dropped. -->
           </ol>
-          <table class="parameters">
-          <tbody>
-              <tr>
-                <th>Parameter</th>
-                <th>Type</th>
-                <th>Nullable</th>
-                <th>Optional</th>
-                <th>Description</th>
-              </tr>
-              <tr>
-                <td class="prmName">data</td>
-                <td class="prmType"><code>Uint8Array</code></td>
-                <td class="prmNullFalse"><span role="img" aria-label=
-                "False">&#10008;</span></td>
-                <td class="prmOptFalse"><span role="img" aria-label=
-                "False">&#10008;</span></td>
-                <td class="prmDesc"></td>
-              </tr>
-            </tbody>
-          </table>
           <div>
-            <em>Return type:</em> <code>Promise&lt;boolean&gt;</code>
+            <em>Return type:</em> <code>WritableStream</code>
           </div>
         </dd>
         <dt><dfn><code>receiveDatagrams</code></dfn></dt>
         <dd>
-          <p>If datagrams have been received since the last call to receiveDatagrams(),
-             return a new promise resolved with all of the received datagrams. </p>
-          <p>If not, return a new promise that will resolve when more datagrams are received,
-              resolved with all datagrams received. </p>
-          <p>If too many datagrams are queued between calls to receiveDatagrams(),
-             the implementation may drop datagrams and replace them with a null value
-             in the sequence of datagrams returned in the next call to receiveDatagrams().
-             One null value may represent many dropped datagrams.<p>
-          <p>receiveDatagrams() may only be called once at a time.
-             If a promised returned from a previous call is still unresolved,
-             the user agent MUST return a new promise rejected with an InvalidStateError. </p>
+          <p>Return the value of the <code><a>[[\ReceivedDatagrams]]</code></a> internal slot.</p>
+          <p>For each datagram received, insert it into <code><a>[[\ReceivedDatagrams]]</code></a>.
+             If too many datagrams are queued because the stream
+             is not being read quickly enough, drop datagrams to avoid queueing.<p>
```

Comment:
How many datagrams should be queued? Which datagrams are being dropped and which are being inserted? Will one always get the most recent datagram or the first added to the queue? I think this is important to specify.

Reply:
I don't think we can give an exact number of packets, but I added some text for guidance as to how to choose (not too big and not too small).